src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    let newState = { ...currentState };
```

Comment:
You are creating a new copy of the currentState on every iteration. This is unnecessary as you are already making a new copy inside each action type. You can remove this line.